react/Gallery.tsx

```diff
@@ -13,37 +29,111 @@ import GalleryLegacy from './GalleryLegacy'
  */
 type GalleryLayoutPropsWithSlots = Omit<GalleryLayoutProps, 'slots'> & Slots
 
+/**
+ * Refactor: This should be provided by the BFF in future as this is inefficient to do
+ *
+ */
+
+export type PreferredSKU =
+  | 'FIRST_AVAILABLE'
+  | 'LAST_AVAILABLE'
+  | 'PRICE_ASC'
+  | 'PRICE_DESC'
+
+const RenderImage = (item: Product, handles: Record<string, string>) => {
+  const [product] = item.items
+
+  const { isMobile } = useDevice()
+  const resizedUrl = changeImageUrlSize(
+    product.images[0].imageUrl,
+    isMobile ? 179 : 280
+  )
+
+  return (
+    <div>
+      <img
```

Comment:
@giretyu can we add width and height attributes to help with CLS?
<img width="677" alt="image" src="https://github.com/TFG-Labs/vtex-tfg-search-result/assets/11704483/d946272e-0fd2-4bd0-8cf5-b7aa2bc15438">